ouroboros-network/src/Ouroboros/Network/Socket.hs

```diff
@@ -272,7 +275,7 @@ beginConnection muxTracer handshakeTracer encodeData decodeData acceptVersion fn
             traceWith muxTracer' $ Mx.MuxTraceHandshakeServerError err
             throwIO err
           Right app -> do
-            traceWith muxTracer' $ Mx.MuxTraceHandshakeEnd
+            traceWith muxTracer' $ Mx.MuxTraceHandshakeServerEnd
```

Comment:
I'm also a bit concerned that we have to use the same mux tracer in the use of the mux, rather than it being all within the mux. It suggests we've not got the boundaries right.

In this case, with the handshake, since the mux knows nothing about the version handshake then it should probably not be in the mux tracer at all.

I realise this is not directly related to the PR at hand, but this PR uses this tracer which is now being used for three things: mux tracing, handshake tracing and delta-q measurement. This is too much.

Reply:
Good point, handshake deserves its own tracer.